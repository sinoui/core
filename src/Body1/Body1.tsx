import { TextContent, typographyPub } from '../utils/typographyPub';

const Body1 = typographyPub('p', 'body1')(TextContent);
export default Body1;
