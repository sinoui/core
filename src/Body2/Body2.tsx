import { TextContent, typographyPub } from '../utils/typographyPub';

const Body2 = typographyPub('p', 'body2')(TextContent);
export default Body2;
