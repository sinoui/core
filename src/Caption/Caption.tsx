import { TextContent, typographyPub } from '../utils/typographyPub';

const Caption = typographyPub('p', 'caption')(TextContent);
export default Caption;
