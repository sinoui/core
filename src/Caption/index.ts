import createTypography from '../Typography/createTypography';

export default createTypography('caption', 'Caption');
