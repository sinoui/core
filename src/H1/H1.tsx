import { TextContent, typographyPub } from '../utils/typographyPub';

const H1 = typographyPub('h1', 'h1')(TextContent);
export default H1;
