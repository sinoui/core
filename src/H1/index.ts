import createTypography from '../Typography/createTypography';

export default createTypography('h1', 'H1');
