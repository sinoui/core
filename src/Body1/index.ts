import Body1 from './Body1';

export default Body1;
