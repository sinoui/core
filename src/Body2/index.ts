import Body2 from './Body2';

export default Body2;
