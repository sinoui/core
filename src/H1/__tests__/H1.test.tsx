import React from 'react';
import renderer from 'react-test-renderer';
// import 'jest-styled-components';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TestWrapper from './TestWrapper';
import Typography from '@sinoui/core/Typography';
import H1 from '@sinoui/core/H1';
import H2 from '@sinoui/core/H2';
import H3 from '@sinoui/core/H3';
import H4 from '@sinoui/core/H4';
import H5 from '@sinoui/core/H5';
import H6 from '@sinoui/core/H6';
import Subtitle1 from '@sinoui/core/Subtitle1';
import Subtitle2 from '@sinoui/core/Subtitle2';
import Body1 from '@sinoui/core/Body1';
import Body2 from '@sinoui/core/Body2';
import Caption from '@sinoui/core/Caption';
import Overline from '@sinoui/core/Overline';

/**
 * 文字排版
 */
describe('文字排版 单元测试', () => {
  afterEach(cleanup);

  test('测试className属性', async () => {
    const { container } = render(
      <TestWrapper>
        <H1>H1</H1>
        <H2>H2</H2>
        <Body2>Body2</Body2>
      </TestWrapper>,
    );

    const text = container.querySelector('.sinoui-typography-h1');
    expect(text).toHaveTextContent('H1');

    const text2 = container.querySelector('.sinoui-typography-h2');
    expect(text2).toHaveTextContent('H2');

    const text3 = container.querySelector('.sinoui-typography-body2');
    expect(text3).toHaveTextContent('Body2');
  });
  test('测试DOM元素', async () => {
    const { container } = render(
      <TestWrapper>
        <H1>H1</H1>
      </TestWrapper>,
    );

    const test = container.querySelector('h1');
    expect(test).toHaveTextContent('H1');
  });
  test('测试设置as属性的DOM元素', async () => {
    const { container } = render(
      <TestWrapper>
        <H1 as="h6">H1</H1>
      </TestWrapper>,
    );

    const test = container.querySelector('h6');
    expect(test).toHaveTextContent('H1');
  });
  test('测试Typography,默认type属性', async () => {
    const { getByTestId } = render(
      <TestWrapper>
        <Typography>H1</Typography>
      </TestWrapper>,
    );

    const test = getByTestId('typography').getAttribute('type');
    expect(test).toEqual('body2');
  });
  test('测试Typography,设置type属性', async () => {
    const { getByTestId } = render(
      <TestWrapper>
        <Typography type="body1">H1</Typography>
      </TestWrapper>,
    );

    const test = getByTestId('typography').getAttribute('type');
    expect(test).toEqual('body1');
  });
  test('测试gutterBottom属性', async () => {
    const { container } = render(
      <TestWrapper>
        <H1>H1</H1>
        <Body2 gutterBottom>Body2</Body2>
        <H1 gutterBottom>H2</H1>
      </TestWrapper>,
    );

    const test = container.querySelector('.sinoui-typography--gutterBottom');
    expect(test).toHaveTextContent('H2');
  });
  test('测试paragraph属性', async () => {
    const { container } = render(
      <TestWrapper>
        <H1>H1</H1>
        <H2 paragraph>H1</H2>
        <Body2 paragraph>Body2</Body2>
      </TestWrapper>,
    );

    const test = container.querySelector('.sinoui-typography--paragraph');
    expect(test).toHaveTextContent('Body2');
  });
});

describe('文字排版 快照测试', () => {
  it('基本文字排版', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <H1>H1</H1>
          <H2>H1</H2>
          <H3>H1</H3>
          <H4>H1</H4>
          <H5>H1</H5>
          <H6>H1</H6>
          <Subtitle1>H1</Subtitle1>
          <Subtitle2>H1</Subtitle2>
          <Body1>H1</Body1>
          <Body2>H1</Body2>
          <Caption>H1</Caption>
          <Overline>H1</Overline>
          <Typography>H1</Typography>
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('设置属性', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <H1 as="h2">H1</H1>
          <H1 align="center">H1</H1>
          <H1 noWrap>H1</H1>
          <H1 color="success">H1</H1>
          <H1 gutterBottom>H1</H1>
          <Body2 paragraph>Body2</Body2>
          <Typography type="body1">Typography</Typography>
          <Typography type="body1" as="p">
            Typography
          </Typography>
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('颜色设置', () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <Typography>Typography</Typography>
          <Typography color="primary">Typography</Typography>
          <Typography color="secondary">Typography</Typography>
          <Typography color="textPrimary">Typography</Typography>
          <Typography color="textSecondary">Typography</Typography>
          <Typography color="error">Typography</Typography>
          <Typography color="warning">Typography</Typography>
          <Typography color="success">Typography</Typography>
          <Typography color="info">Typography</Typography>
        </TestWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
