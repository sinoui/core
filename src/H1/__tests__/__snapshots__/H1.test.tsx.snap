// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`文字排版 快照测试 基本文字排版 1`] = `
Array [
  <h1
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-h1"
    color="textPrimary"
  >
    H1
  </h1>,
  <h2
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-h2"
    color="textPrimary"
  >
    H1
  </h2>,
  <h3
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-h3"
    color="textPrimary"
  >
    H1
  </h3>,
  <h4
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-h4"
    color="textPrimary"
  >
    H1
  </h4>,
  <h5
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-h5"
    color="textPrimary"
  >
    H1
  </h5>,
  <h6
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-h6"
    color="textPrimary"
  >
    H1
  </h6>,
  <h6
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-subtitle1"
    color="textPrimary"
  >
    H1
  </h6>,
  <h6
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-subtitle2"
    color="textPrimary"
  >
    H1
  </h6>,
  <p
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-body1"
    color="textPrimary"
  >
    H1
  </p>,
  <p
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-body2"
    color="textPrimary"
  >
    H1
  </p>,
  <p
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-caption"
    color="textPrimary"
  >
    H1
  </p>,
  <p
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-overline"
    color="textPrimary"
  >
    H1
  </p>,
  <div
    className="typographyObj__Text-sc-1f9on5o-0 Typography__TextWrapper-sc-12agh80-0 eOqWRf sinoui-typography"
    color="textPrimary"
    data-testid="typography"
    type="body2"
  >
    H1
  </div>,
]
`;

exports[`文字排版 快照测试 设置属性 1`] = `
Array [
  <h2
    className="typographyObj__Text-sc-1f9on5o-0 hdagpW sinoui-typography-h1"
    color="textPrimary"
  >
    H1
  </h2>,
  <h1
    className="typographyObj__Text-sc-1f9on5o-0 kpkbsC sinoui-typography-h1"
    color="textPrimary"
  >
    H1
  </h1>,
  <h1
    className="typographyObj__Text-sc-1f9on5o-0 ioNPnn sinoui-typography-h1"
    color="textPrimary"
  >
    H1
  </h1>,
  <h1
    className="typographyObj__Text-sc-1f9on5o-0 eUjDii sinoui-typography-h1"
    color="success"
  >
    H1
  </h1>,
  <h1
    className="typographyObj__Text-sc-1f9on5o-0 dTjhLA sinoui-typography-h1 sinoui-typography--gutterBottom"
    color="textPrimary"
  >
    H1
  </h1>,
  <p
    className="typographyObj__Text-sc-1f9on5o-0 foFpwc sinoui-typography-body2 sinoui-typography--paragraph"
    color="textPrimary"
  >
    Body2
  </p>,
  <div
    className="typographyObj__Text-sc-1f9on5o-0 Typography__TextWrapper-sc-12agh80-0 eOqWRf sinoui-typography"
    color="textPrimary"
    data-testid="typography"
    type="body1"
  >
    Typography
  </div>,
  <p
    className="typographyObj__Text-sc-1f9on5o-0 Typography__TextWrapper-sc-12agh80-0 eOqWRf sinoui-typography"
    color="textPrimary"
    data-testid="typography"
    type="body1"
  >
    Typography
  </p>,
]
`;
